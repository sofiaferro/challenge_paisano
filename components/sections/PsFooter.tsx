import React from 'react';
import styled from 'styled-components';

import { theme } from '@/styles/theme';
import texts from '@/styles/texts';

import Image from 'next/image';
import logo from '@/images/logo.png';

import PsHorizontalDivider from '../molecules/PsHorizontalDividerLarge';

// styles
const { secondBg, primaryText } = theme;
const { primaryBody, secondaryCaption, secondaryBody } = texts;

const Container = styled.div`
  ${secondBg};
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 2em;
`;
const ImageWrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding-top: 2em;
  padding-bottom: 2em;
  @media (min-width: 680px) {
    justify-content: center;
  }
`;

const Text = styled.p`
  ${primaryBody};
  ${primaryText};
  font-weight: 400;
  padding-bottom: 2em;
  text-align: center;
  @media (max-width: 680px) {
    width: 50%;
    ${secondaryBody}
    text-align: left;
  }
`;

const Copyright = styled.p`
  ${secondaryCaption};
  text-align: center;
`;

const PsFooter = () => {
  return (
    <Container>
      <ImageWrapper>
        <Image src={logo} alt='NFTPaisanos logo' />
      </ImageWrapper>
      <Text>The New Creative Economy.</Text>
      <PsHorizontalDivider />
      <Copyright>Created with ❤ by sofi 👩‍💻👾</Copyright>
    </Container>
  );
};

export default PsFooter;
