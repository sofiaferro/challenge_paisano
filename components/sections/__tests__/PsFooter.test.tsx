import React from 'react';
import renderer from 'react-test-renderer';

import { describe, expect, it } from '@jest/globals';

import PsFooter from '../PsFooter';
describe('PsFooter', () => {
  it('renders the component', () => {
    const tree = renderer.create(<PsFooter />);
    expect(tree).toMatchSnapshot();
  });
});
