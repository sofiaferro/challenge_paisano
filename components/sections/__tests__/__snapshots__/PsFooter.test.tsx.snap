// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsFooter renders the component 1`] = `
<div
  className="sc-6b502cdb-0 fdAMRV"
>
  <div
    className="sc-6b502cdb-1 dDBJSW"
  >
    <img
      alt="NFTPaisanos logo"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
  </div>
  <p
    className="sc-6b502cdb-2 loeXDe"
  >
    The New Creative Economy.
  </p>
  <div
    className="sc-e5ce1a5b-0 kzTqjN"
  >
    <img
      alt="Horizontal divider"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
  </div>
  <p
    className="sc-6b502cdb-3 dkXOCx"
  >
    Created with ❤ by sofi 👩‍💻👾
  </p>
</div>
`;
